### PR TITLE
Fix links.json schema_type in TDR plugin tests (#2363)

### PR DESCRIPTION
**Pull request by noah-aviel-dove**
Author (before primary review)

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue
- [x] Added `DCP/1` label <sub>or this PR does not target an `hca/*` branch</sub>
- [x] Created issue to track porting these changes <sub>or this PR does not need porting</sub> 
- [x] Added `[r]` prefix at start of commit title <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label <sub>or this PR does not require reindexing</sub>
- [x] Freebies are blocked on this PR <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in chain <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR <sub>or this PR is not chained to another PR</sub>
- [x] Ran `make requirements_update` <sub>or this PR does not change requirements*.txt</sub>
- [x] Mentioned `make requirements` in UPGRADING.rst <sub>or this PR does not change requirements*.txt</sub>
- [x] Documented upgrade of personal deployments in UPGRADING.rst <sub>or this PR does not require upgrading</sub>
- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>

Primary reviewer (before pushing merge commit)

- [x] Checked `reindex` label and `[r]` prefix of commit title
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Build passes in `sandbox` <sub>or added `no sandbox` label</sub>
- [x] Reindexed `sandbox` <sub>or this PR does not require reindexing</sub>
- [x] Added PR reference to merge commit
- [x] Moved linked issue to Merged
- [ ] Pushed merge commit to Github

Primary reviewer (after pushing merge commit)

- [ ] Moved freebies to Merged column <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened chain <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Commented on demo expectations <sub>or labeled as `no demo`</sub>
- [ ] Pushed merge commit to Gitlab <sub>or this changes can be pushed later, together with another PR</sub>
- [ ] Reindexed `dev` <sub>or this PR does not require reindexing</sub>
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Unassign reviewer from PR

### CODE REVIEW

**Comment by noah-aviel-dove** [comment]
To confirm the schema used by Terra:
```
SELECT DISTINCT JSON_EXTRACT_SCALAR(content, '$.schema_type')
FROM hca_ucsc_files_20201001_optimusb.links
------
row | f0_
1 | links
```

**Review by amarjandu — APPROVED** [review]
Looks good, just a reminder to run integration tests. :)

File: test/service/test_tdr.py
Comment (hannes-ucsc) [inline test/service/test_tdr.py:326]: I think this comment will confuse us in the future. What is V1 and why do we need to point out a thing of the past?

Comment (noah-aviel-dove) [reply test/service/test_tdr.py:326]: "DCP v1." 

To save us from writing out the entire contents of the expected output from the TDR plugin's bundle emulation, the tests load an old bundle and mutate it to reflect the changes in DCP v2. There are many several similar comments throughout the `_convert_metadata` and `_convert_manifest` methods, explaining what inconsistencies are being addressed.

Comment (hannes-ucsc) [reply test/service/test_tdr.py:326]: Makes sense. "DCP/1" would be more descriptive.

https://github.com/DataBiosphere/azul/issues/2374

File: test/service/test_tdr.py
Comment (hannes-ucsc) [inline test/service/test_tdr.py:326]: "DCP/1" here and everywhere else, obviously.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
Use DCP/1 instead of v1 consistently. Do that substitution in a separate first commit. Subsequent commit should introduce the semantic changes.

**Comment by noah-aviel-dove** [comment]
@hannes-ucsc Travis build passed previously but appears stuck after the migration.

**Review by hannes-ucsc — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
I'll merge this when blocker PR #2278 lands, since there may be conflicts between the two PR branches.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]
Please resolve conflicts. I tried but it's not trivial so I'd rather leave it to the author.

**Review by hannes-ucsc — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
Sandbox build was not freshly rebased, the base was one commit behind HEAD develeop but that commit was only documentation changes.

**Comment by hannes-ucsc** [comment]
I don't think this has an observable effect on the index. So there is nothing to demo.

**Pull request merged**
